### PR TITLE
Adding token auth and model api to workflow and https

### PR DESCRIPTION
**Pull request by udaij12**
## Description

Adding token auth tests to workflow and https
Added model control tests to https 

Workflow does not need model control tests as the model registered with the workflow.

### CODE REVIEW

**Review by agunapal — COMMENTED** [review]
Do we have tests for the behavior when you enable token-auth?

**Comment by udaij12** [comment]
The two tests added are with token auth enabled

**Comment by agunapal** [comment]
> The two tests added are with token auth enabled

Can you please add tests for the reverse case. i.e without a token , they should fail

**Comment by udaij12** [comment]
Added two tests without token which fail

**Review by agunapal — APPROVED** [review]
minor comment. LGTM otherwise

File: test/pytest/test_token_authorization.py [hunk 1]
```diff
@@ -89,6 +97,39 @@ def setup_torchserve_expiration():
...
+        ("model_name", "mnist"),
+        ("url", "mnist.mar"),
+        ("initial_workers", "1"),
+        ("synchronous", "true"),
+    )
+    response = requests.post(
+        "https://localhost:8081/models", params=params, headers=header, verify=False
+    )
+    file_content = Path(f"{CURR_DIR}/key_file.json").read_text()
+    print(file_content)
```
[the remark this answers is not in the document]
Comment (udaij12) [reply test/pytest/test_token_authorization.py:126]: prints the key file so that in case the tests fail can see the expected key

**Pull request merged**